pallets/loans/src/lib.rs

```diff
@@ -428,18 +433,20 @@ pub mod module {
         /// Called by substrate on block initialization.
         /// Our initialization function is fallible, but that's not allowed.
         fn on_initialize(block_number: T::BlockNumber) -> frame_support::weights::Weight {
-            match <Pallet<T>>::accrue_interest() {
-                Ok(()) => 0,
-                Err(err) => {
-                    // This should never happen...
-                    log::info!(
-                        "Could not initialize block!!! {:#?} {:#?}",
-                        block_number,
-                        err
-                    );
-                    0
+            with_transaction(|| {
+                match <Pallet<T>>::accrue_interest() {
+                    Ok(()) => TransactionOutcome::Commit(1000),
```

Comment:
what does 1000 mean ?

Reply:
It should be the total weight consumed in the `accrue_interest`. Consider that there is no benchmark build for `accrue_interest`, I use 1000 for now.